### PR TITLE
Moved pending deployment

### PR DESCRIPTION
**Pull request by danielskinstad**
Question: do we need to check anything else, and is it in the correct place? Is comparing the artifact type from the deployment data necessary?

### CODE REVIEW

File: core/src/mender-client.c [hunk 1]
```diff
@@ -1092,6 +1092,11 @@ mender_client_check_deployment(mender_api_deployment_data_t **deployment_data) {
     cJSON_AddStringToObject(mender_client_deployment_data, "artifact_name", deployment->artifact_name);
     cJSON_AddArrayToObject(mender_client_deployment_data, "types");
 
+    assert(NULL != mender_client_deployment_data);
```
Comment (vpodzime) [inline core/src/mender-client.c:1095]: This is guaranteed by the `if` a couple lines above, right?

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise.

**Comment by vpodzime** [comment]
> Is comparing the artifact type from the deployment data necessary?

No, I didn't quite understand what the purpose of this was. We definitely need to make sure the artifact type is supported by some available update module after a reboot, but that should/does happen in a different place.


> do we need to check anything else, and is it in the correct place?

As suspicious as your change looks I believe it's correct. The logic should jump into a the update function which should either continue or abort an update/deployment or check for a new one.

File: core/src/mender-client.c (hunk 1 above)
Comment (danielskinstad) [reply core/src/mender-client.c]: Yep

**Comment by vpodzime** [comment]
> @vpodzime Where is going to happen? Do we have a task covering that? Otherwise I'd say we need to add it in this PR (the JIRA description was not very clear to me)

We should do it [here](https://github.com/mendersoftware/mender-mcu/blob/feature-update_modules/core/src/mender-client.c#L1197), I think. But let me create a ticket for it, we need to also properly handle the case when there is a new deployment with an artifact of an unsupported type. The current code would jump into the `FAILURE` state and try to call a callback of a `NULL` module.

**Comment by vpodzime** [comment]
> But let me create a ticket for it

https://northerntech.atlassian.net/browse/MEN-7524

**Pull request merged**